javascript-bank/account.js

```diff
@@ -0,0 +1,53 @@
+/* exported Account */
+function Account(number, holder) {
+  this.holder = holder;
+  this.number = number;
+  this.transactions = [];
+}
+
+Account.prototype.deposit = function (amount) {
+  if (Number.isInteger(amount) && amount > 0) {
+
+    var accountDeposit = new Transaction(
+      this.type = 'deposit',
+      this.amount = amount
+    );
```

Comment:
The `Transaction` constructor takes the `type` and `amount` as _arguments_, meaning you don't need to assign them to `this.anything`. In fact, on lines 12-13, `this` refers to the current `Account` object, not the `Transaction` object you're  creating.

---

![image](https://user-images.githubusercontent.com/16143771/151905468-1559d792-560b-43dd-a238-f99e3ecbe04e.png)